Week3/app.js

```diff
@@ -0,0 +1,239 @@
+   */
+
+  function newstring(str = '') {
+    let secondstring = '';
+    for (let i = 0; i < str.length; i++) {
+      if (str.charCodeAt(i) >= 97 && str.charCodeAt(i) <= 122) {
+        secondstring = secondstring.concat(str[i].toUpperCase());
+      } else if (str.charCodeAt(i) >= 65 && str.charCodeAt(i) <= 90) {
+        secondstring = secondstring.concat(str[i].toLowerCase());
+      } else {
+        secondstring = secondstring.concat(str[i]);
+      }
+    }
+    return secondstring;
+  }
+  console.log(newstring('The Quick Brown Fox'));
+}
+
+//More JavaScript
+
+// 1. Create a function that takes 3 arguments and returns the sum of the these arguments.
+
+function sum(arr) {
+  return arr.reduce(function(a, b, c) {
+    return a + b + c;
+  }, 0);
+}
+
+let numbers = [3, 4, 5];
+
+console.log(sum(numbers));
+
+// 2. Create a function named colorCar that receives a color, and prints out, 'a red car' for example.
+const colorCar = color => `a ${color} car`;
+console.log(colorCar('red'));
+
+// 3. Create an object and a function that takes the object as a parameter and prints out all of its properties and values.
+
+const carproperties = {
+  name: 'mercedes',
+  productiondate: '2018',
+  price: '50.000 euro',
+};
+
+const print = () => {
+  for (const properties in carproperties) {
+    console.log(`mercedes ${properties}: ${carproperties[properties]}`);
+  }
+};
+print();
+
+// 4.Create a function named vehicleType that receives a color, and a code, 1 for car, 2 for motorbike. And prints 'a blue motorbike' for example when called as vehicleType("blue", 2)
+
+const vehicleType = (color, code) => {
+  if (code === 1) {
+    console.log(`a ${color} car`);
+  } else if (code === 2) {
+    console.log(`a ${color} motorbike`);
+  } else {
+    console.log('nothing');
+  }
+};
+vehicleType('yellow', 2);
+
+/* 5. Can you write the following without the if statement, but with just as a single line with console.log(...);?
+ if (3 === 3) {
+    console.log("yes");
+  } else {
+    console.log("no");
+  }
+ */
+console.log(3 === 3 ? `yes` : `no`);
+
+// 6.Create a function called vehicle, like before, but takes another parameter called age, so that vehicle("blue", 1, 5) prints 'a blue used car'
+
+const vehicle = (color, code, age) => {
+  const newa = age === 0 ? `new` : `used`;
+
+  if (code === 1) {
+    console.log(`a ${color} ${newa} car`);
+  } else if (code === 2) {
+    console.log(`a ${color} ${newa} motorbike`);
+  }
+};
+vehicle('blue', 1, 1);
+
+// 7. Make a list of vehicles, you can add "motorbike", "caravan", "bike", or more.
+const vehicles = ['motorbike', 'caravan', 'bike', 'car'];
+
+//8. How do you get the third element from that list?
+
+console.log(vehicles[2]);
+
+//9. Change the function vehicle to use the list of question 7. So that vehicle("green", 3, 1) prints "a green new bike".
+
+let vehic = (color, code, age) => {
+  age === 1 || age === 0 ? (age = 'new') : (age = 'used');
```

Comment:
Ah, this works yeah! But re-assigning variables that are passed through the function is not recommended:) just so you know

Reply:
yeah, I will use your recommended next time